open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -49,7 +72,432 @@ def get_available_docker_host(self, req_count, hackathon):
+        """
+        Set endpoint configuration which is need in creating Azure VM
+
+        :param service_name: the name of cloud service
+        :type service_name: str|unicode
+
+        :param hackathon_id: the id of hackathon
+        :type hackathon_id: integer
+
+        :return: the endpoint configuration set
+        :rtype: class 'azure.servicemanagement.ConfigurationSet'
+        """
+        used_port_set = self.__get_used_public_port_set(service_name, hackathon_id)
+        docker_port = 4243
+        ssh_port = 9000
+        while docker_port in used_port_set:
+            docker_port += 1
+        while ssh_port in used_port_set:
+            ssh_port += 1
+        assert docker_port < 40000 and ssh_port < 40000
+        endpoint_config = ConfigurationSet()
+        endpoint_config.configuration_set_type = 'NetworkConfiguration'
+        endpoint1 = ConfigurationSetInputEndpoint(name='docker', protocol='tcp',
+                                                  port=docker_port, local_port='4243',
+                                                  load_balanced_endpoint_set_name=None,
+                                                  enable_direct_server_return=False)
+        endpoint2 = ConfigurationSetInputEndpoint(name='SSH', protocol='tcp',
+                                                  port=ssh_port, local_port='22',
+                                                  load_balanced_endpoint_set_name=None,
+                                                  enable_direct_server_return=False)
+        endpoint_config.input_endpoints.input_endpoints.append(endpoint1)
+        endpoint_config.input_endpoints.input_endpoints.append(endpoint2)
+        return endpoint_config
+
+    def __deployment_exists(self, service_name, deployment_slot, hackathon_id):
+        """
+        Check whether the deployment of a cloud service exists in the Azure account which is related to hackathon_id
+
+        :param service_name:the name of service
+        :type service_name: str|unicode
+
+        :param deployment_slot: the slot of deployment
+        :type deployment_slot: str|unicode
+
+        :param hackathon_id: the id of hackathon
+        :type hackathon_id: integer
+
+        :return: True if there does exist the cloud service
+                 False if not found such a service
+        :rtype: bool
+
+        :raise: raise exception when something went wrong with Azure api
+                "Something wrong with checking deployment of service:service_name"
+        :type: Exception
+        """
+        sms = self.__get_sms_object(hackathon_id)
+        try:
+            sms.get_deployment_by_slot(service_name, deployment_slot)
+            return True
+        except Exception as e:
+            if e.message == 'Not found (Not Found)':
+                return False
+            else:
+                raise Exception('Something wrong with checking deployment of service:' % service_name)
+
+    def __find_all_hackathon_id(self):
```

Comment:
Cannot find where method `__find_all_hackathon_id` is called? What's the purpose of it?